polaris-service/src/main/java/org/apache/polaris/service/auth/DefaultOAuth2ApiService.java

```diff
@@ -39,8 +40,8 @@
  * Default implementation of the {@link OAuth2ApiService} that generates a JWT token for the client
  * if the client secret matches.
  */
-@JsonTypeName("default")
-public class DefaultOAuth2ApiService implements OAuth2ApiService, HasMetaStoreManagerFactory {
+@Named("default")
```

Comment:
`@Named` has a couple drawbacks.
Using specialized `@Qualifier`s using `AnnotationLiteral`s to select the configured type via `Instance.select()` is the most flexible but also the safest way.

Reply:
Using `@Named` allows the existing configuration parsing to be used. HK2 does allow for accessing the different instances (e.g., using `IterableProvider`), but that's not really exclusive of using the `@Named` annotation

Reply:
There's `Instance.select()` in CDI for such cases - in combination with `@Qualifier` and `AnnotationLiteral` (mentioned elsewehere IIRC).